pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/MultiStageBrokerRequestHandler.java

```diff
@@ -191,6 +192,11 @@ protected BrokerResponse handleRequest(long requestId, String query, @Nullable S
     try {
       queryResults = _queryDispatcher.submitAndReduce(requestContext, dispatchableSubPlan, queryTimeoutMs, queryOptions,
           stageIdStatsMap);
+    } catch (TimeoutException e) {
```

Comment:
Can you verify if it can throw `TimeoutException`?

Reply:
Yes actually, the query dispatcher throws a timeout exception.